src/opera/pge/dist_s1/dist_s1_pge.py

```diff
@@ -271,6 +331,61 @@ def _validate_rtcs(self):
         self.__validate_rtc_homogeneity(current_matches)  # Per ADT: Baseline can be heterogeneous
         self.__validate_rtc_ordering(all_rtcs)
         self.__validate_co_and_cross_polarizations(all_rtcs)
+        self.__validate_no_duplicates(baseline_matches + current_matches)
+
+    def _validate_previous_product(self):
+        """
+        Run validations for previous product input.
+
+        Checks:
+        1. All files exist
+        2. All files are nonempty
+        3. All files are .tif files
+        4. Correct number of files (one for each TIFF layer) provided
+        5. All files are named conformant to DIST-S1 product specification
+        6. Exact set of layers provided (ie, no duplicates)
+        """
+        sas_config = self.runconfig.sas_config
+        previous_product = sas_config["run_config"].get("pre_dist_s1_product", [])
```

Comment:
Its a bit confusing that `pre_dist_s1_product` is actually a list of products plural as defined by the schema. So I would rather see this attriubte named `previous_products` to make the following code clearer.

Reply:
It's a single product in the form of multiple .tif files, so maybe `previous_product_files`?